tests/rptest/tests/cloud_retention_test.py

```diff
@@ -289,26 +294,40 @@ def test_retention_with_node_failures(self, cloud_storage_type):
 
         # Write fast enough that in the test's runtime, we would certainly
         # overshoot the retention size substantially if retention wasn't working.
-        write_rate = (self.retention_bytes * 4.0) / runtime
+        write_rate = int(self.retention_bytes * 4.0) // runtime
 
         # Write enough data to last the runtime.
         msg_size = 128
-        msg_count = write_rate * runtime // msg_size
+        msg_count = int(write_rate * runtime // msg_size)
 
-        producer = KgoVerifierProducer(self.test_context,
-                                       self.redpanda,
-                                       self.topic,
-                                       msg_count=msg_count,
-                                       msg_size=msg_size,
-                                       rate_limit_bps=write_rate)
+        producer = KgoVerifierProducer(
+            self.test_context,
+            self.redpanda,
+            self.topic,
+            msg_count=msg_count,
+            msg_size=msg_size,
+            # Use small batches to avoid overshooting segment size target too much
+            batch_max_bytes=max(msg_size * 2, 512),
+            rate_limit_bps=write_rate)
         producer.start()
 
+        # Wait for some substantial production to happen before we start
+        # inspecting cloud storage
+        producer.wait_for_acks(msg_count / 2,
+                               timeout_sec=runtime,
+                               backoff_sec=5)
+
         def cloud_log_size() -> int:
             s3_snapshot = BucketView(self.redpanda, topics=self.topics)
             if not s3_snapshot.is_ntp_in_manifest(self.topic, 0):
                 self.logger.debug(f"No manifest present yet")
                 return 0
 
+            if s3_snapshot.manifest_for_ntp(topic=self.topic, partition=0).get(
+                    'start_offset', 0) <= 0:
+                self.logger.debug(f"Manifest not prefix-truncated yet")
+                return 0
```

Comment:
I might be missing something but it's a little surprising to see this as a condition to get the size: if we haven't truncated yet, we still have a size, no?

Reply:
Yeah, this function's return value was never really used, so the name doesn't make sense.  I've changed it.